pyproject.toml

```diff
@@ -0,0 +1,14 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[tool.coverage.paths]
+source = ["src", "*/site-packages"]
+
+[tool.coverage.run]
+branch = true
+relative_files = true
+
+[tool.coverage.report]
+show_missing = true
+exclude_lines = ["if __name__ == .__main__.:", "pragma: no-cover", "@abstract", "def __repr__"]
```

Comment:
`build-system` section is alternative of `setup` of `setup.py`. When there is no `setup.py`,  and when using `setuptools` for build, it is to be a same configuraiton.

A file `pyproject.toml` is used for test, not to supporting `pyproject.toml` in users project.
These lines are adeed for testing purpose.

Reply:
Testing of the pyproject.toml support? I think the extra changes in this PR were a little confusing otherwise so it was more difficult to review by the maintainer.

Reply:
I'm well aware of what pyproject.toml is and its purpose. I was only trying to say that it isn't clearly mentioned in any comments or in the PR description what the extra configuration in this pyproject.toml is doing. The `coverage` configuration is different from the existing configuration in `setup.cfg` (which is not removed in this PR). You also define a build system where previously there wasn't one so it isn't clear why one is needed.

I'm also a little confused because the tests in `push.yml` and in `make tests` don't explicitly handle `pyproject.toml` so by putting it here in the root of the repository I assume all tests are now using this `pyproject.toml`. However, that doesn't test the absence of a `pyproject.toml` (by using the existing setup.cfg). Shouldn't both cases be tested? Perhaps with a subdirectory for different test configuration cases? One with a setup.cfg and one with a pyproject.toml?